core/src/main/java/org/wso2/msf4j/internal/MicroservicesRegistryImpl.java

```diff
@@ -92,8 +92,13 @@ public void addService(String basePath, Object service) {
     }
 
     public void removeService(Object service) {
-        services.remove(service);
-        updateMetadata();
+        Path path = service.getClass().getAnnotation(Path.class);
```

Comment:
Shall we check for null for the service object input and handle it ? Otherwise, that could lead to a possible NPE. WDYT?